src/watch/watch.go

```diff
@@ -26,7 +24,7 @@ const debounceInterval = 50 * time.Millisecond
 // Watch starts watching the sources of the given labels for changes and triggers
 // rebuilds whenever they change.
 // It never returns successfully, it will either watch forever or die.
-func Watch(state *core.BuildState, labels []core.BuildLabel, run bool) {
+func Watch(state *core.BuildState, labels core.BuildLabels, run bool, runWatchededBuild func(args []string)) {
```

Comment:
I don't think it's ideal to pass command-line arguments back and forth as an interface; among other things this package doesn't know how to reconstruct them.

I think the callback function should probably take a list of the build labels that this package thinks it should build, and hook back into the build process with those (assuming that's even necessary... maybe we just always build the same set every time?).